### PR TITLE
Updates to `AddClaimantModal`

### PR DESCRIPTION
**Pull request by jcq**
Connects #14153

### Description
This adds functionality to `AddClaimantModal` — implementing debounce on searchable dropdown and async fetching from backend API.

### Acceptance Criteria
- [ ] Code compiles correctly
- [ ] Adds debounce functionality
- [ ] Pulls select options via async fetch from backend
- [ ] Saves claimant `participant_id`
- [ ] Displays added claimant in radio list
- [ ] Supports removing claimant from list
- [ ] Fleshes out feature test

### CODE REVIEW

**Comment by jcq** [comment]
Note: will need to update story to include local async fn for generating names

**Comment by jcq** [comment]
## Updated Screenshots:

Screen Shot 2020-05-22 at 15 31 19
Screen Shot 2020-05-22 at 15 31 29

File: client/app/intake/components/AddClaimantModal.jsx [hunk 1]
```diff
@@ -36,22 +45,39 @@ export const AddClaimantModal = ({ onCancel, onSubmit }) => {
...
   ];
 
   return (
-    <Modal title={ADD_CLAIMANT_MODAL_TITLE} buttons={buttons} closeHandler={onCancel}>
+    <Modal title={ADD_CLAIMANT_MODAL_TITLE} buttons={buttons} closeHandler={onCancel} id="add_claimant_modal">
       <div>
         <ReactMarkdown source={ADD_CLAIMANT_MODAL_DESCRIPTION} />
       </div>
-      <SearchableDropdown label="Claimant's name" options={dropdownOpts} onChange={handleChange} value={claimant} />
+      <SearchableDropdown
```
[the remark this answers is not in the document]
Comment (jcq) [reply client/app/intake/components/AddClaimantModal.jsx:58]: @rutvigupta-design Just wanted to flag this for you

[the remark this answers is not in the document]
Comment (rutvigupta-design) [reply client/app/intake/components/AddClaimantModal.jsx:58]: I'd love to take a look at the screenshots, if possible!

File: spec/feature/intake/intake_spec.rb [hunk 2]
```diff
@@ -538,7 +538,10 @@
 
     context "Adding new claimant" do
```
[the remark this answers is not in the document]
Comment (jcq) [reply spec/feature/intake/intake_spec.rb]: Moving the tests in there as I fully support keeping that stuff together. Good flag!

File: spec/feature/intake/intake_spec.rb [hunk 3]
```diff
@@ -538,7 +538,10 @@
...
       let(:hlr) { create(:higher_level_review, veteran_file_number: "12341234") }
-      let(:intake) { HigherLevelReviewIntake.new(veteran_file_number: "12341234", user: current_user) }
+      let(:intake) { AppealIntake.new(veteran_file_number: "12341234", user: current_user) }
```
[the remark this answers is not in the document]
Comment (jcq) [reply spec/feature/intake/intake_spec.rb]: Must have been copied from somewhere and not removed. Definitely will remedy.

**Comment by jcq** [comment]
> Hm, while playing around with this and prepping for demo, I might've stumbled on a flaw in how we're using this component. The XHR and the React state contain results, but the `<Select>` is only showing those with an exact (case-insensitive) substring match.
> 
> 
> 
> This makes the dropdown pretty intolerant of typos, and defeats some of the fuzzy matching special sauce 😅

Oof, good flag. Taking a look.

**Comment by jcq** [comment]
@nanotone I think the update I just pushed should cover the issue you raised (it will now return all options that it received from the backend).

**Comment by jcq** [comment]
> When testing locally, it looks like I was able to fire off these 8 XHRs within a span of about 400 milliseconds. Does that square with how you understand the debouncing strategy?

Nope, looks like there must have been a regression on that. I'll look into it and follow up with another PR. Good catch!

**Comment by nanotone** [comment]
> Nope, looks like there must have been a regression on that. I'll look into it and follow up with another PR. Good catch!

Food for thought: this feels like a great opportunity for a frontend test. It's confined to one or a few components, and is the kind of test that benefits from the snappiness of ditching rspec+capybara. I see that Jest comes with some nifty timer mocks, too.

**Pull request merged**